packages/dx-demo-shell/src/demo-viewer/source-code.jsx

```diff
@@ -64,12 +72,22 @@ export class SourceCode extends React.PureComponent {
       }).join('\n');
   }
 
+  applySpecialCodeOptions() {
+    this.foldBlockStartLines.forEach((lineNumber) => {
+      this.codeMirror.foldCode(CodeMirror.Pos(lineNumber, 0));
+    });
+    this.importantLines.forEach((lineNumber) => {
+      this.codeMirror.addLineClass(lineNumber, 'background', 'CodeMirror-important-line');
+    });
+  }
+
   render() {
-    const sourceCode = this.prepareSourceCode();
+    const { themeName, sectionName, demoName } = this.props;
+    const sourceCode = this.prepareSourceCode(themeName, sectionName, demoName);
```

Comment:
Even though now `foldBlockStartLines` and `importantLines` are reset on each `render` it is still not pure as modifying instance fields is a side effect.

However it will most like be automatically resolved when `componentWillReceiveProps` is gone.